packages/react-components/src/shared/src/useControllableState.ts

```diff
@@ -74,7 +74,7 @@ export interface ControllableStateOptions<T> {
  * The goal of this hook is to seemlessly support "controlled" and "uncontrolled" component behaviors.
  * This is achieved by abstracting the state and updating a state value only when a prop is considered "uncontrolled".
  */
-export function useControllableState<T>(controlledValue: T, initialValue: T, defaultValue: T, { onChange }: ControllableStateOptions<T> = {}): [T, (maybeState: T) => void, boolean] {
+export function useControllableState<T>(controlledValue: T | undefined, initialValue: T | undefined, defaultValue: T | undefined, { onChange }: ControllableStateOptions<T> = {}): [T, (maybeState: T) => void, boolean] {
```

Comment:
Same here for `null` ?

Reply:
L'idée c'est que T ici je veux pas qu'il infere un type avec undefined dedans. Par exemple, si on a `selectedKey?: string| null` je veux que T soit string | null, pas string| null | undefined. Je fais ça parce que le return type de useControllableState c'est un state en T aussi, mais celui là sera pas undefined, puisqu'on lui assigne une default value.

donc une variable `selectedKey?: string| null` passé à useControllableState va être useControllableState<string | null>